hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemStoreLAB.java

```diff
@@ -191,11 +192,18 @@ private Chunk getOrMakeChunk() {
       // No current chunk, so we want to allocate one. We race
       // against other allocators to CAS in an uninitialized chunk
       // (which is cheap to allocate)
-      c = (chunkPool != null) ? chunkPool.getChunk() : new Chunk(chunkSize);
+      if(chunkPool != null) {
+        c = chunkPool.getChunk();
+      } else {
+        // HBASE-14921: 555 is here till it is decided whether ChunkPool is always on
+        c = new Chunk(chunkSize, 555);
+        c.init();
+      }
+
       if (curChunk.compareAndSet(null, c)) {
         // we won race - now we need to actually do the expensive
         // allocation step
-        c.init();
+
         this.chunkQueue.add(c);
         return c;
       } else if (chunkPool != null) {
```

Comment:
Again, convoluted. Can we just split the chunkPool != null and chunkPool == null paths up-front?